pkg/apis/pipeline/v1beta1/pipeline_validation.go

```diff
@@ -78,6 +78,8 @@ func (ps *PipelineSpec) Validate(ctx context.Context) (errs *apis.FieldError) {
 	errs = errs.Also(validateMatrix(ctx, ps.Tasks).ViaField("tasks"))
 	errs = errs.Also(validateMatrix(ctx, ps.Finally).ViaField("finally"))
 	errs = errs.Also(validateResultsFromMatrixedPipelineTasksNotConsumed(ps.Tasks, ps.Finally))
+	// Emit warnings for deprecated fields
+	errs = errs.Also(warnPipelineFieldsDeprecation(ctx, ps))
```

Comment:
I think individual validation functions should be responsible for emitting warnings, rather than one overall function responsible for emitting all deprecation warnings.